src/components/Calculator.js

```diff
@@ -1,42 +1,67 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
-// eslint-disable-next-line react/prefer-stateless-function
+// eslint-disable-next-line
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((prevState) => calculate(prevState, e.target.innerText));
+  }
+
   render() {
+    const { total, next, operation } = this.state;
     return (
       <div id="calculator">
         <div id="input-area">
-          <input id="input-field" dir="rtl" />
+          <span id="input-field">
+            {total}
+            {operation}
+            {next}
+          </span>
         </div>
         <div id="button-area">
           <div className="row row-1">
-            <button type="button" className="calc-button first-line">AC</button>
-            <button type="button" className="calc-button first-line">+/-</button>
-            <button type="button" className="calc-button first-line">%</button>
-            <button type="button" className="orange-button calc-button first-line">/</button>
+            <button onClick={this.handleClick} type="button" className="calc-button first-line">AC</button>
+            <button onClick={this.handleClick} type="button" className="calc-button first-line">+/-</button>
+            <button onClick={this.handleClick} type="button" className="calc-button first-line">%</button>
```

Comment:
- [ ] When I try any number modulo zero the app crash. Kindly fix the issue maybe try displaying a message like when the user tries dividing by zero.